pkg/transforms/filter_test.go

```diff
@@ -149,6 +182,64 @@ func TestFilterByValueDescriptor(t *testing.T) {
 	assert.Len(t, res.(models.Event).Readings, 1, "Event should have one reading")
 }
 
+func TestFilterOutByValueDescriptor(t *testing.T) {
+	f1 := NewFilter([]string{descriptor1})
+	f1.FilterOut = true
+	f12 := NewFilter([]string{descriptor1, descriptor2})
+	f12.FilterOut = true
+
+	// event with a value descriptor 1
+	event1 := models.Event{
+		Device: devID1,
+	}
+	event1.Readings = append(event1.Readings, models.Reading{Name: descriptor1})
+
+	// event with a value descriptor 2
+	event2 := models.Event{}
+	event2.Readings = append(event2.Readings, models.Reading{Name: descriptor2})
+
+	// event with a value descriptor 1 and another 2
+	event12 := models.Event{}
+	event12.Readings = append(event12.Readings, models.Reading{Name: descriptor1})
+	event12.Readings = append(event12.Readings, models.Reading{Name: descriptor2})
+
+	// event with a value descriptor 2
+	event3 := models.Event{}
+	event3.Readings = append(event3.Readings, models.Reading{Name: descriptor3})
+
+	continuePipeline, res := f1.FilterByValueDescriptor(context)
+	assert.False(t, continuePipeline, "Pipeline should stop since no parameter was passed")
+	assert.EqualError(t, res.(error), "no Event Received")
+
+	continuePipeline, res = f1.FilterByValueDescriptor(context, event1)
+	assert.True(t, continuePipeline, "Pipeline should continue")
+	assert.Len(t, res.(models.Event).Readings, 0, "Event should have no readings")
+
+	continuePipeline, res = f1.FilterByValueDescriptor(context, event2)
+	assert.True(t, continuePipeline, "Pipeline should continue")
+	assert.Len(t, res.(models.Event).Readings, 1, "Event should have one reading")
+
+	continuePipeline, res = f1.FilterByValueDescriptor(context, event12)
+	assert.True(t, continuePipeline, "Event should be filtered out")
+	assert.Len(t, res.(models.Event).Readings, 1, "Event should have one reading")
+
+	continuePipeline, res = f12.FilterByValueDescriptor(context, event1)
+	assert.True(t, continuePipeline, "Pipeline should continue")
+	assert.Len(t, res.(models.Event).Readings, 0, "Event should have one reading")
```

Comment:
Interesting, if we have filtered out all the readings, shouldn't we filter out the whole `Event`? I.e. why pass forward an Event with no readings. @rsdmike ??

Reply:
that makes sense. It probably wouldn't be very helpful sending a blank event.

Reply:
that makes sense. It probably wouldn't be very helpful sending a blank event.

Reply:
@rsdmike , what are your thoughts here?

Reply:
I added a separate issue to deal with this.